aten/src/ATen/native/mps/operations/GridSampler.mm

```diff
@@ -0,0 +1,156 @@
+          if (static_cast<GridSamplerInterpolation>(interpolation_mode) == GridSamplerInterpolation::Nearest) {
+            outputTensor = [mpsGraph sampleGridWithSourceTensor: inputTensor
+                                               coordinateTensor: gridTensor
+                                                         layout: inputTensorLayout
+                                           normalizeCoordinates: TRUE
+                                            relativeCoordinates: FALSE
+                                                   alignCorners: align_corners
+                                                    paddingMode: paddingMode
+                                            nearestRoundingMode: MPSGraphResizeNearestRoundingModeRoundToEven
+                                                  constantValue: 0.0f
+                                                           name: nil];
+          } else {
+            outputTensor = [mpsGraph sampleGridWithSourceTensor: inputTensor
+                                               coordinateTensor: gridTensor
+                                                         layout: inputTensorLayout
+                                           normalizeCoordinates: TRUE
+                                            relativeCoordinates: FALSE
+                                                   alignCorners: align_corners
+                                                    paddingMode: paddingMode
+                                                   samplingMode: samplingMode
+                                                  constantValue: 0.0f
+                                                           name: nil];
+          }
+
+          newCachedGraph->inputTensor_ = inputTensor;
+          newCachedGraph->gridTensor_ = gridTensor;
+          newCachedGraph->outputTensor_ = outputTensor;
+        }
+        return newCachedGraph;
+      });
+      cachedGraph = static_cast<CachedGraph *>(tmpCachedGraph);
+    }
+
+    Placeholder inputPlaceholder = Placeholder(cachedGraph->inputTensor_, input);
+    Placeholder gridPlaceholder = Placeholder(cachedGraph->gridTensor_, grid);
+    Placeholder outputPlaceholder = Placeholder(cachedGraph->outputTensor_, output);
+
+
+    NSDictionary<MPSGraphTensor*, MPSGraphTensorData*>* feeds = @{
+      inputPlaceholder.getMPSGraphTensor() : inputPlaceholder.getMPSGraphTensorData(),
+      gridPlaceholder.getMPSGraphTensor() : gridPlaceholder.getMPSGraphTensorData()
+    };
+    NSDictionary<MPSGraphTensor*, MPSGraphTensorData*>* results = @{
+      outputPlaceholder.getMPSGraphTensor() : outputPlaceholder.getMPSGraphTensorData()
+    };
+
+    runMPSGraph(stream, cachedGraph->graph(), feeds, results);
+  }
+#endif // defined(__MAC_13_2)
+}
+
+Tensor grid_sampler_2d_mps(const Tensor& input, const Tensor& grid,
+                           int64_t interpolation_mode, int64_t padding_mode,
+                           bool align_corners) {
+#if defined(__MAC_13_2)
```

Comment:
This means that GridSample is likely not available in any of our binary builds... :/

Reply:
Interesting. There is a test added for grid sampler which should fail if that is the case. I'll take a look